fmriprep/interfaces/itk.py

```diff
@@ -8,15 +8,166 @@
 
 
 """
+import os
 import numpy as np
 import nibabel as nb
+
+from joblib import Parallel, delayed
+from mimetypes import guess_type
+from tempfile import TemporaryDirectory
+
 from niworkflows.nipype.utils.filemanip import fname_presuffix
 from niworkflows.nipype.interfaces.base import (
-    traits, TraitedSpec, BaseInterfaceInputSpec, File)
+    traits, TraitedSpec, BaseInterfaceInputSpec, File, InputMultiPath, OutputMultiPath)
 
+from niworkflows.nipype.interfaces.ants.resampling import ApplyTransformsInputSpec
 from niworkflows.interfaces.base import SimpleInterface
 
 
+class MCFLIRT2ITKInputSpec(BaseInterfaceInputSpec):
+    in_files = InputMultiPath(File(exists=True), mandatory=True,
+                              desc='list of MAT files from MCFLIRT')
+    in_reference = File(exists=True, mandatory=True,
+                        desc='input image for spatial reference')
+    in_source = File(exists=True, mandatory=True,
+                     desc='input image for spatial source')
+    nprocs = traits.Int(1, usedefault=True, nohash=True,
+                        desc='number of parallel processes')
+
+
+class MCFLIRT2ITKOutputSpec(TraitedSpec):
+    out_file = File(desc='the output ITKTransform file')
+
+
+class MCFLIRT2ITK(SimpleInterface):
+
+    """
+    Convert a list of MAT files from MCFLIRT into an ITK Transform file.
+    """
+    input_spec = MCFLIRT2ITKInputSpec
+    output_spec = MCFLIRT2ITKOutputSpec
+
+    def _run_interface(self, runtime):
+        tmp_folder = TemporaryDirectory()
+        parallel = Parallel(n_jobs=self.inputs.nprocs)
+        itk_outs = parallel(delayed(_mat2itk)(
+            in_mat, self.inputs.in_reference, self.inputs.in_source, i,
+            tmp_folder.name) for i, in_mat in enumerate(self.inputs.in_files))
+        tmp_folder.cleanup()
```

Comment:
It's fine how it is, but I find it a little cleaner to use `TemporaryDirectory()` as a context manager:

```Python
parallel = Parallel(n_jobs=self.inputs.nprocs)
with TemporaryDirectory() as tmp_folder:
    itk_outs = parallel(delayed(_mat2itk)(
        in_mat, self.inputs.in_reference, self.inputs.in_source, i,
        tmp_folder.name) for i, in_mat in enumerate(self.inputs.in_files))
```